core/src/main/scripts/mutationalSignatures.py

```diff
@@ -0,0 +1,369 @@
+#!/usr/bin/env python3
+"""
+Author(s): Matthijs Pon
+Company/Institution: The Hyve
+Email: matthijs@thehyve.nl
+"""
+
+# imports
+import argparse
+import glob
+import os
+import shutil
+import subprocess
+import pandas as pd
+
+from rpy2 import robjects
+from SigProfilerMatrixGenerator import install as genInstall
+from SigProfilerMatrixGenerator.scripts import SigProfilerMatrixGeneratorFunc as matGen
+
+
+def parse_args():
+    """Parse command line arguments
+    
+    :return: object, ArgumentParser with parsed objects
+    """
+    parser = argparse.ArgumentParser(
+        description="Extract mutational signatures from a cBioPortal study and generate output in cBioPortal format.",
+    )
+    subparsers = parser.add_subparsers(dest="algorithm",
+                                       help="select algorithm for mutational signature extraction")
+    parser.add_argument("study_path", metavar="study-path",
+                        help="cBioPortal study path")
+    parser.add_argument("out_dir", metavar="out-dir",  # TODO this can be an optional parameter and default can be study path?
+                        help="output directory")
+    parser.add_argument("--install-genome", dest="install_genome", action='store_true',
+                        help="install the given genome for SigProfilerMatrixGenerator (only required on first run "
+                             "per genome)")
+    parser.add_argument("--tmp-dir", dest="tmp_dir", metavar="'./tmp/'", default="./tmp",
+                        help="temporary directory for files, directory will be deleted after runtime (default: ./tmp)")
+
+    ts_parser = subparsers.add_parser("tempoSig")
+    ts_parser.add_argument("--temposig", dest="temposig_loc", metavar="'./tempoSig.R'", default=None,
+                           help="location of tempoSig.R script, only necessary if file cannot be automatically found "
+                                "or installed")
+    ts_parser.add_argument("--cosmic", metavar="[2/3]", default=2, type=int,
+                           help="cosmic version to use in tempoSig (default: 2)")
+    ts_parser.add_argument("--nperm", metavar="n", type=int, default=1000,
+                           help="number of permutations for p-value estimation (default: 1000)")
+    
+    # This parser is unnecessary for now, but provided for easier algorithm implementation
+    sp_parser = subparsers.add_parser("SigProfiler")
+
+    args = parser.parse_args()
+    return args
+
+
+def load_yaml(file):
+    """Simple load a yaml file. Only accepts key: value pairs, no multi-line statements
+    
+    :param file: object, open yaml file 
+    :return: dict, yaml key-value pairs
+    """
+    yaml_cont = {}
+    for line in file:
+        split_line = line.strip().split(":")
+        key = split_line[0].strip()
+        val = "".join(split_line[1:]).strip()
+        yaml_cont[key] = val
```

Comment:
Why not:
```suggestion
    yaml_cont = yaml.full_load(open(file))
```
? It would require importing yaml.

Reply:
some meta files turned out to contain extra colon chars in the description... 
Normally I would also opt for `yaml.full_load` but this is more robust